src/Dev/TestAssetStore.php

```diff
@@ -194,6 +204,65 @@ public function getDefaultConflictResolution($variant)
         return parent::getDefaultConflictResolution($variant);
     }
 
+    /**
+     * Stores grants directly on TestAssetStore to avoid Session dependency in tests
+     *
+     * @param string $filename
+     * @param string $hash
+     */
+    public function grant($filename, $hash)
+    {
+        $member = Security::getCurrentUser();
+        $fileID = $this->getFileID($filename, $hash);
+
+        self::$grantedFiles[$member ? $member->ID : 0][$fileID] = true;
+    }
+
+    /**
+     * Revokes grants directly on TestAssetStore to avoid Session dependency in tests
+     *
+     * @param string $filename
+     * @param string $hash
+     */
+    public function revoke($filename, $hash)
+    {
+        $member = Security::getCurrentUser();
+        $fileID = $this->getFileID($filename, $hash);
+
+        self::$grantedFiles[$member ? $member->ID : 0][$fileID] = false;
+    }
+
+    /**
+     * Fetches grants directly from TestAssetStore to avoid Session dependency in tests
+     *
+     * @param string $filename
+     * @param string $hash
+     */
+    public function isGranted($fileID)
```

Comment:
Why not just call the protected method with the same name?

Reply:
You're totally right 🤦 I've dropped out the other overloaded methods and adjusted `isGranted()` to call its parent.